docs/rate_limiter.md

```diff
@@ -0,0 +1,107 @@
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Rate Limiter
+
+Rate limiter manages the rate at which requests are scheduled on
+model instances by Triton. The rate limiter operates across all
+models loaded in Triton to allow *cross-model prioritization*.
+
+In absence of rate limiting (--rate-limit=off), Triton schedules
+execution of request (or set of requests when using batching) as
+soon as a model instance is available. This behavior is typically
+best suited for performance. However, there can be cases 
+where running all the models simultaneously places excessive load 
+on the server. For instance, model execution on some 
+frameworks dynamically allocate memory. Running all such models
+simultaneously may lead to system going out-of-memory.
+
+Rate limiter allows to post-pone the inference execution on some
+model instances such that not all of them runs simultaneously. 
+The model priorities are used to decide which model instance
+to schedule next. 
+
+## Using Rate Limiter
+
+To enable rate limiting users must set `--rate-limit` server
+cli option. For more information, consult usage of the option
+emitted by `tritonserver --help`.
+
+The rate limiter is controlled by rate limiter configuration
+described in [rate limiter config](model_configuration.md#rate-limiter-config).
+The rate limiter configuration includes 
+[resources](model_configuration.md#resources) and
+[priority](model_configuration.md#priority) for the instances
+defined by the instance group.
+
+### Resources
+
+Resources are just names with a corresponding count. By default
+instance uses no rate-limiter resources. By listing a 
+resource/count the instance indicates that it requests that
+many resources to be available on the instance device before
+it can be allowed to execute. When under execution the specified
+many resources are allocated to the instance only to be released
+when the execution is over. The available number of resources
+is, by default, the max across all instances that list that resource
+(and so must be adjusted each time an instance is added/removed).
+For example, assume three loaded model instances A, B and C each 
+specifying the following resource counts for a single device:
+
+```
+A: [R1: 4, R2: 4]
+B: [R2: 5, R3: 10, R4: 5]
+C: [R1: 1, R3: 7, R4: 2]
+```
+
+By default, the server will select following available resource
+counts in the system:
+
+```
+R1: 4
+R2: 5
+R3: 10
+R4: 7
+```
+
+These values ensure that all model instances can be successfully
+scheduled. The default for a resource can be overridden by giving
+it explicitly on command-line using `--rate-limit-resource` option.
+`tritonserver --help` will provide with more detailed usage
+instructions.
+
```

Comment:
Here I would say that, by default, resources are per-device and resource requirements for a model instance are enforced against the corresponding resources associated with the device where the instance runs. Then describe how global resources can also be specified in instance group and that global resources are shared across the entire server.